src/orchestrator/streaming.rs

```diff
@@ -21,6 +21,8 @@ use std::{collections::HashMap, pin::Pin, sync::Arc, time::Duration};
 
 use aggregator::Aggregator;
 use futures::{future::try_join_all, Stream, StreamExt, TryStreamExt};
+
+use hyper::HeaderMap;
```

Comment:
I noticed this import is different from the `HeaderMap` in other files (i.e. `orchestrator.rs` used `axum::http::HeaderMap`), it seems `axum`, `reqwest` and `hyper` all export the same `HeaderMap` from `http` so this shouldnt cause any correctness issues but maybe for consistency/removing confusion, could we import from the same crate?